(venv) D:\PROJECT\Simulasi-Kasus-Lahan>python scripts/test_processing.py

(venv) D:\PROJECT\Simulasi-Kasus-Lahan>python scripts/test_visualization.py

(venv) D:\PROJECT\Simulasi-Kasus-Lahan>python run_analysis.py
python: can't open file 'D:\\PROJECT\\Simulasi-Kasus-Lahan\\run_analysis.py': [Errno 2] No such file or directory

(venv) D:\PROJECT\Simulasi-Kasus-Lahan>jupyter notebook notebooks/exploration.ipynb
[I 2025-09-04 22:41:37.333 ServerApp] Extension package jupyter_lsp took 0.3455s to import
[I 2025-09-04 22:41:37.579 ServerApp] Extension package jupyter_server_terminals took 0.2454s to import
[I 2025-09-04 22:41:38.176 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2025-09-04 22:41:38.187 ServerApp] jupyter_server_terminals | extension was successfully linked.
[I 2025-09-04 22:41:38.203 ServerApp] jupyterlab | extension was successfully linked.
[I 2025-09-04 22:41:38.216 ServerApp] notebook | extension was successfully linked.
[I 2025-09-04 22:41:38.229 ServerApp] Writing Jupyter server cookie secret to C:\Users\ThinkPad\AppData\Roaming\jupyter\runtime\jupyter_cookie_secret
[I 2025-09-04 22:41:39.366 ServerApp] notebook_shim | extension was successfully linked.
[I 2025-09-04 22:41:39.525 ServerApp] notebook_shim | extension was successfully loaded.
[I 2025-09-04 22:41:39.528 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2025-09-04 22:41:39.529 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2025-09-04 22:41:39.533 LabApp] JupyterLab extension loaded from D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyterlab
[I 2025-09-04 22:41:39.533 LabApp] JupyterLab application directory is D:\PROJECT\Simulasi-Kasus-Lahan\venv\share\jupyter\lab
[I 2025-09-04 22:41:39.534 LabApp] Extension Manager is 'pypi'.
[I 2025-09-04 22:41:40.054 ServerApp] jupyterlab | extension was successfully loaded.
[I 2025-09-04 22:41:40.062 ServerApp] notebook | extension was successfully loaded.
[I 2025-09-04 22:41:40.064 ServerApp] Serving notebooks from local directory: D:\PROJECT\Simulasi-Kasus-Lahan\notebooks
[I 2025-09-04 22:41:40.064 ServerApp] Jupyter Server 2.17.0 is running at:
[I 2025-09-04 22:41:40.064 ServerApp] http://localhost:8888/tree?token=889b690c86c3860782f1e6b6c9110f0d9e868cb545534d51
[I 2025-09-04 22:41:40.064 ServerApp]     http://127.0.0.1:8888/tree?token=889b690c86c3860782f1e6b6c9110f0d9e868cb545534d51
[I 2025-09-04 22:41:40.065 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 2025-09-04 22:41:40.133 ServerApp]

    To access the server, open this file in a browser:
        file:///C:/Users/ThinkPad/AppData/Roaming/jupyter/runtime/jpserver-26740-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/tree?token=889b690c86c3860782f1e6b6c9110f0d9e868cb545534d51
        http://127.0.0.1:8888/tree?token=889b690c86c3860782f1e6b6c9110f0d9e868cb545534d51
[I 2025-09-04 22:41:40.334 ServerApp] Skipped non-installed server(s): basedpyright, bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyrefly, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
[I 2025-09-04 22:41:41.676 JupyterNotebookApp] 302 GET /tree/exploration.ipynb?token=[secret] (6772c7afbf5c40c0841afae6aaccabe9@::1) 3.36ms
[W 2025-09-04 22:41:49.493 ServerApp] 400 GET /api/contents/exploration.ipynb?type=notebook&content=1&hash=1&contentProviderId=undefined&1757000509016 (::1): Unreadable Notebook: D:\PROJECT\Simulasi-Kasus-Lahan\notebooks\exploration.ipynb NotJSONError("Notebook does not appear to be JSON: '# %%\\nimport pandas as pd\\nimport geopa...")
[W 2025-09-04 22:41:49.494 ServerApp] wrote error: 'Unreadable Notebook: D:\\PROJECT\\Simulasi-Kasus-Lahan\\notebooks\\exploration.ipynb NotJSONError("Notebook does not appear to be JSON: \'# %%%%\\\\nimport pandas as pd\\\\nimport geopa...")'
    Traceback (most recent call last):
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\tornado\web.py", line 1848, in _execute
        result = await result
                 ^^^^^^^^^^^^
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyter_server\auth\decorator.py", line 73, in inner
        return await out
               ^^^^^^^^^
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyter_server\services\contents\handlers.py", line 156, in get
        model = await ensure_async(
                ^^^^^^^^^^^^^^^^^^^
        ...<7 lines>...
        )
        ^
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyter_core\utils\__init__.py", line 197, in ensure_async
        result = await obj
                 ^^^^^^^^^
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyter_server\services\contents\filemanager.py", line 924, in get
        model = await self._notebook_model(path, content=content, require_hash=require_hash)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyter_server\services\contents\filemanager.py", line 870, in _notebook_model
        nb, bytes_content = await self._read_notebook(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
            os_path, as_version=4, capture_validation_error=validation_error, raw=True
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        )
        ^
      File "D:\PROJECT\Simulasi-Kasus-Lahan\venv\Lib\site-packages\jupyter_server\services\contents\fileio.py", line 497, in _read_notebook
        raise HTTPError(
        ...<2 lines>...
        )
    tornado.web.HTTPError: HTTP 400: Bad Request (Unreadable Notebook: D:\PROJECT\Simulasi-Kasus-Lahan\notebooks\exploration.ipynb NotJSONError("Notebook does not appear to be JSON: '# %%\\nimport pandas as pd\\nimport geopa..."))
[W 2025-09-04 22:41:49.719 ServerApp] 400 GET /api/contents/exploration.ipynb?type=notebook&content=1&hash=1&contentProviderId=undefined&1757000509016 (6772c7afbf5c40c0841afae6aaccabe9@::1) 364.52ms referer=http://localhost:8888/notebooks/exploration.ipynb